### ISSUE TITLE
[explorer] updateAddrBalances: attempted to subtract {X} coin(s) from address {A} which only has {Y} coin(s)

### ISSUE THREAD

**Issue by vantuz-subhuman** [input-output-hk/cardano-sl#3409]
Hello!

While syncing the explorer node I often see these error messages, like:
```
[node:ERROR:ThreadId 331] [2018-08-15 07:31:36.17 UTC] updateAddrBalances: attempted to subtract 114437037947 coin(s) from address DdzFFzCqrhtB66YkKzouHqeTAznkyCL7xfwAZvrgqkGiqhkjBcLSV3D8hCn9Ug7xBrpXwHEukKoWJq1w6cdh1a2Nt6fGp3M5tVocVPSV which only has 33113406445 coin(s)
[node:ERROR:ThreadId 331] [2018-08-15 07:31:36.18 UTC] updateAddrBalances: attempted to subtract 114422786900 coin(s) from unknown address DdzFFzCqrhsoVf3kMyQtNLkQ9EAwnTtmCXn325RFHfiiGV2WNCWmHYEYx3wq6UDpi1uduSeR44ce8qtKmrr7TAa9qJNVe5jMKqjGC5qb
[node:ERROR:ThreadId 331] [2018-08-15 07:31:36.19 UTC] updateAddrBalances: attempted to subtract 114382896933 coin(s) from address DdzFFzCqrhtB66YkKzouHqeTAznkyCL7xfwAZvrgqkGiqhkjBcLSV3D8hCn9Ug7xBrpXwHEukKoWJq1w6cdh1a2Nt6fGp3M5tVocVPSV which only has 33113406445 coin(s)
```

Node keeps syncing after reporting it, and looks like everything works fine.

Is this ok? How there can be a historical attempt to subtract something invalid? If a transaction is invalid - it wouldn't be included in the chain; and if a transaction is in the chain - shouldn't it be **guaranteed** to be valid?

Doesn't it mean that the explorer node accounts addresses wrong?

For example, if we check out a summary for one of these addresses and compare it with alternative explorer:

1. Official one shows 108 transactions and balance of 7,217.944985 coins - http://cardanoexplorer.com/api/addresses/summary/DdzFFzCqrhtB66YkKzouHqeTAznkyCL7xfwAZvrgqkGiqhkjBcLSV3D8hCn9Ug7xBrpXwHEukKoWJq1w6cdh1a2Nt6fGp3M5tVocVPSV

2. Alternative one shows 3880 transactions and balance of 134,461,796.862673 coins - https://adatracker.com/address/DdzFFzCqrhtB66YkKzouHqeTAznkyCL7xfwAZvrgqkGiqhkjBcLSV3D8hCn9Ug7xBrpXwHEukKoWJq1w6cdh1a2Nt6fGp3M5tVocVPSV

It does look strange.

**Comment by vantuz-subhuman** [comment]
+ @gufmar

**Comment by vantuz-subhuman** [comment]
This issue is **critical**, because it's the **OFFICIAL EXPLORER**, and the whole community relies on it for checking the validity of information.

**Comment by brkphp** [comment]
> Are there any updates on this? Our node can't finish syncing as it crashes every few minutes due to this error.

I have the same problem, unfortunately.

**Comment by FreekPaans** [comment]
Same here, only for me the node doesn't crash, but it starts using a lot of resources (cpu,mem,disk), and then I need to kill it.

**Comment by michdr** [comment]
This problem still occurs. Any solution for it?

**Comment by Victoryfan** [comment]
Yeah, I also have this problem, how to solve it？Increase the RAM?
![image](https://user-images.githubusercontent.com/25604389/61283246-68df5580-a7ef-11e9-8f69-c348bc7cb56a.png)

**Comment by erikd** [comment]
I am currently working on debugging an issue closely related to this.

**Comment by Victoryfan** [comment]
> I am currently working on debugging an issue closely related to this.

It occured when sycn data via explorer-mainnet, and i send an email to you to ask for your help. I wish you can help me. Thanks in advance.
